core/domain/topic_domain_test.py

```diff
@@ -755,7 +754,7 @@ def test_update_thumbnail_filename(self) -> None:
             self.topic.update_thumbnail_filename('img.svg')
 
         # Save the dummy image to the filesystem to be used as thumbnail.
-        with python_utils.open_file(  # type: ignore[no-untyped-call]
+        with utils.open_file( # type: ignore[call-overload]
```

Comment:
Why is the `# type: ignore[call-overload]` needed?

ditto below

Reply:
Here in the use of open_file , the encoding is set to None ( which is disregarded by default as encoding should always be specified ) , Thus i needed to skip the mypy checks for these 2 particular lines where encoding is set to None .

Reply:
Can we either enforce the encoding here or allow `None` in the utils?

Reply:
It is very rare to see encoding being specified as None specifically in oppia codebase . Only here in this particular case it was specified . So in my opinion it would be better to let it stay like this . In future , if there are more used cases for encoding as None , we can then add None type to encoding in utils . What's your say ?

Reply:
Why does the encoding have to be None in the first place? If the encoding doesn't have to be None then we shouldn't have this special case at all, especially given that this is just a test file. It'd be better if it was whatever it is we use regularly in the main codebase.

Reply:
@seanlip I have no idea why encoding is set here to None . I just tried to cover these new introduced errors without modifying much of the original code written there . If you want i can try changing the encoding there ( None ) and see if it's work fine as it should .

Reply:
Please try changes here that will help get rid of the ignores.

Reply:
Should i just add the encoding ( None ) as optional to the open_file function in utils.py

Reply:
hey i added the None to the encoding . It will remove the need to skip those erros . PTAL https://github.com/oppia/oppia/pull/14728/commits/3ff110477b87c2caffe0d351b7ca1ac21d3a593c

Reply:
Can we fix this without allowing `None`, can you just try to force some different encoding in this file?

Reply:
Agreed with @vojtechjelinek. Ideally you would save and read the SVG in the same way that we do in the main codebase (outside the test files).

Reply:
@vojtechjelinek @seanlip I will to implement what you're asking . Also please do provide me with any leads, ideas if you come across any regarding solving this issue

Reply:
yes , and that's the whole point . Assets like image, audio, svg needs to be saved in raw format to preserve their data .

Reply:
yes limiting None to binary types is doable and seems more accurate !!

Reply:
If None is a standard option -- I don't quite understand why, in the entire codebase, we are using None in only a single test (and not even in the main code). Could you shed some light on that?

> The reason why we are not using any encoding is because we don't want to modify the original arrangement of data and that is very crusial 

I don't really buy this. I think you might want to provide links to references to substantiate your arguments.

Reply:
@seanlip No. certainly it's not a single place where we are using None encoding . Check this output : 
```bash
╰─ git grep "encoding=None" | cat
core/controllers/android_e2e_config.py:            'rb', encoding=None) as f:
core/controllers/android_e2e_config.py:            'rb', encoding=None) as f:
core/controllers/android_e2e_config.py:            encoding=None) as f:
core/controllers/base_test.py:            'rb', encoding=None
core/controllers/blog_dashboard_test.py:            encoding=None
core/controllers/blog_dashboard_test.py:            encoding=None
core/controllers/creator_dashboard_test.py:        'rb', encoding=None
core/controllers/domain_objects_validator_test.py:                encoding=None
core/controllers/editor_test.py:            golden_zip_filepath, 'rb', encoding=None) as f:
core/controllers/editor_test.py:            'rb', encoding=None
core/controllers/editor_test.py:            'rb', encoding=None
core/controllers/question_editor_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'), 'rb', encoding=None
core/controllers/resources_test.py:            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'), 'rb', encoding=None
core/controllers/resources_test.py:            encoding=None
core/controllers/resources_test.py:            encoding=None
core/controllers/resources_test.py:            encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/resources_test.py:            'rb', encoding=None
core/controllers/suggestion_test.py:            'rb', encoding=None
core/controllers/suggestion_test.py:            encoding=None
core/controllers/suggestion_test.py:            'rb', encoding=None
core/controllers/suggestion_test.py:            'rb', encoding=None
core/controllers/suggestion_test.py:            encoding=None) as f:
core/controllers/suggestion_test.py:            'rb', encoding=None
core/controllers/topic_editor_test.py:            encoding=None
core/controllers/topic_editor_test.py:            encoding=None
core/controllers/topic_editor_test.py:            encoding=None
core/controllers/topics_and_skills_dashboard_test.py:            'rb', encoding=None
core/controllers/topics_and_skills_dashboard_test.py:            'rb', encoding=None
core/controllers/topics_and_skills_dashboard_test.py:            encoding=None
core/controllers/topics_and_skills_dashboard_test.py:            'rb', encoding=None
core/domain/exp_services_test.py:            encoding=None) as f:
core/domain/exp_services_test.py:            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'), 'rb', encoding=None
core/domain/exp_services_test.py:            os.path.join(feconf.TESTS_DATA_DIR, 'cafe.mp3'), 'rb', encoding=None
core/domain/exp_services_test.py:            encoding=None
core/domain/exp_services_test.py:            encoding=None) as f:
core/domain/fs_services_test.py:            os.path.join(feconf.TESTS_DATA_DIR, 'img.png'), 'rb', encoding=None
core/domain/fs_services_test.py:            encoding=None) as f:
core/domain/fs_services_test.py:            encoding=None) as f:
core/domain/fs_services_test.py:            encoding=None) as f:
core/domain/html_validation_service_test.py:            encoding=None) as f:
core/domain/html_validation_service_test.py:            encoding=None) as f:
core/domain/html_validation_service_test.py:            encoding=None) as f:
core/domain/image_services_test.py:            'rb', encoding=None) as f:
core/domain/image_services_test.py:            'rb', encoding=None) as f:
core/domain/image_services_test.py:            'rb', encoding=None) as f:
core/domain/image_validation_services_test.py:            'rb', encoding=None) as f:
core/domain/rte_component_registry_test.py:                generated_image_filepath, 'rb', encoding=None) as f:
core/domain/story_domain_test.py:            encoding=None) as f:
core/domain/story_domain_test.py:            'rb', encoding=None) as f:
core/domain/story_services_test.py:            'rb', encoding=None) as f:
core/domain/story_services_test.py:            'rb', encoding=None) as f:
core/domain/suggestion_registry_test.py:            'rb', encoding=None) as f:
core/domain/suggestion_registry_test.py:            encoding=None) as f:
core/domain/topic_domain_test.py:            'rb', encoding=None) as f:
core/domain/topic_domain_test.py:            encoding=None) as f:
core/domain/topic_services_test.py:            encoding=None) as f:
core/domain/topic_services_test.py:            'rb', encoding=None) as f:
core/domain/user_services_test.py:            expected_gravatar_filepath, 'rb', encoding=None) as f:
core/python_utils.py:        encoding=None, bind_and_activate=True):
extensions/interactions/base_test.py:            with utils.open_file(png_file, 'rb', encoding=None) as f:
scripts/build.py:    with utils.open_file(filepath, 'rb', encoding=None) as f:
scripts/rtl_css.py:            css_file = python_utils.open_file(file_path, 'rb', encoding=None)
scripts/rtl_css.py:                rtl_file_path, 'rb', encoding=None).read()
scripts/rtl_css_test.py:        def mock_open(filepath, mode, encoding=None): # pylint: disable=unused-argument
scripts/rtl_css_test.py:        def errorred_mock_open(filepath, mode, encoding=None): # pylint: disable=unused-argument
```

Reply:
An earlier part of this comment thread said:

> It is very rare to see encoding being specified as None specifically in oppia codebase . Only here in this particular case it was specified . 

I made my comments based on that. Are you saying that None is used elsewhere in general? If so then my perspective on this changes.